packages/components/src/Dialog/Layout/DialogLayout.tsx

```diff
@@ -88,9 +107,7 @@ export const DialogLayout: FC<DialogLayoutProps> = ({
   headerDetail,
   isLoading,
 }) => {
-  const dialogFooter = (footer || footerSecondary) && (
-    <DialogFooter secondary={footerSecondary}>{footer}</DialogFooter>
-  )
+  const dialogFooter = constructDialogFooter(footer, footerSecondary)
```

Comment:
Given the simplicity of this logic I'd probably just inline it here:

```suggestion
  const dialogFooter = footer ? (
    <DialogFooter secondary={footerSecondary}>{footer}</DialogFooter>
  ) : null
```